src/rest/DataQueryEndpoints.js

```diff
@@ -62,13 +64,14 @@ module.exports = (historicalAdapter, streamFetcher, volumeLogger = new VolumeLog
             })
         } else {
             const dataPoints = []
-            historicalAdapter.getLast(
+            const streamingData = storage.fetchLatest(
                 req.params.id,
                 partition,
                 count,
-                dataPoints.push.bind(dataPoints),
-                onDataFetchDone(res, dataPoints, wrapperOption.toLowerCase(), contentOption.toLowerCase(), volumeLogger),
             )
+            streamingData.on('error', onDataFetchDone(res))
+            streamingData.on('data', dataPoints.push.bind(dataPoints))
+            streamingData.on('end', onDataFetchDone(res, dataPoints, wrapperOption.toLowerCase(), contentOption.toLowerCase(), volumeLogger))
```

Comment:
Are we sure the two arguments are right here? Seems like this is passing strings (`toLowerCase()`) for content and volumeLogger?